pkg/syncer/metadatasyncer.go

```diff
@@ -2886,6 +2886,90 @@ func policyQuotaCRAdded(obj interface{}) {
+func policyQuotaCRDeleted(obj interface{}) {
+	ctx, log := logger.GetNewContextWithLogger()
+	// Verify object received.
+	var policyQuotaObj storagepolicyusagev1alpha1.StoragePolicyQuota
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.(*unstructured.Unstructured).Object,
+		&policyQuotaObj)
+	if err != nil {
+		log.Errorf("policyQuotaCRDeleted: failed to cast object %+v to %s. Error: %v", obj,
+			cnsoperatorv1alpha1.CnsStoragePolicyQuotaSingular, err)
+		return
+	}
+	namespace := policyQuotaObj.Namespace
+	storagePolicyId := policyQuotaObj.Spec.StoragePolicyId
+	// Get storage classes associated with storage policy id of StoragePolicyQuota CR added
+	config, err := k8s.GetKubeConfig(ctx)
+	if err != nil {
+		log.Errorf("policyQuotaCRDeleted: Failed to get KubeConfig. err: %v", err)
+		return
+	}
+	k8sClient, err := clientset.NewForConfig(config)
+	if err != nil {
+		log.Errorf("policyQuotaCRDeleted: Failed to create kubernetes client. Err: %+v", err)
+		return
+	}
+	storageQuotaClient, err := k8s.NewClientForGroup(ctx, config, cnsoperatorv1alpha1.GroupName)
+	if err != nil {
+		log.Errorf("policyQuotaCRDeleted: Failed to create CnsOperator client. Err: %+v", err)
+		return
+	}
+	storageClassList, err := k8sClient.StorageV1().StorageClasses().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		log.Errorf("policyQuotaCRDeleted: Failed to list storageclasses. Err: %+v", err)
+		return
+	}
+	policyUsageList := &storagepolicyusagev1alpha1.StoragePolicyUsageList{}
+	err = storageQuotaClient.List(ctx, policyUsageList, &client.ListOptions{
+		Namespace: namespace,
+	})
+	if err != nil {
+		log.Errorf("policyQuotaCRDeleted: Failed to list %v CRs associated in namespace %v. Err: %+v",
+			cnsoperatorv1alpha1.CnsStoragePolicyUsageSingular, namespace, err)
+		return
+	}
+	// For each storage class associated with storage policy id of StoragePolicyQuota CR,
+	// check if StoragePolicyUsage CR with resource type PVC exists.
+	// If found, delete the one with all the specified parameters
+	for _, sc := range storageClassList.Items {
+		if sc.Parameters[scParamStoragePolicyID] == storagePolicyId {
+			policyUsageInstanceName := sc.Name + "-" + "pvc-usage"
+			for _, usage := range policyUsageList.Items {
+				if usage.Spec.StoragePolicyId == storagePolicyId &&
+					usage.Spec.StorageClassName == sc.Name {
+					if usage.Spec.ResourceKind == ResourceKindPVC {
+						policyUsageCR := storagepolicyusagev1alpha1.StoragePolicyUsage{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      policyUsageInstanceName,
+								Namespace: namespace,
+							},
+						}
+						err := storageQuotaClient.Delete(ctx, &policyUsageCR)
+						if err != nil {
+							if apierrors.IsNotFound(err) {
+								log.Infof("policyQuotaCRDeleted: StoragePolicyUsageCR instance %q/%q already deleted. "+
+									"Returning success for the delete operation", namespace, policyUsageInstanceName)
+								return
```

Comment:
we have two SC per policy? Do we need to continue here instead of return?

Reply:
Done, addressed in the latest commit